### PR TITLE
yoav - Fix pivot

### PR DESCRIPTION
**Pull request by greenblitz4590**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
+                RobotConstants.General.Motors.SIGNAL_FREQUENCY_HERTZ,
+                motor.getPosition(),
+                motor.getVelocity(),
+                motor.getMotorVoltage(),
+                motor.getSupplyCurrent(),
+                motor.getDeviceTemp(),
+                motor.getAcceleration(),
+                motor.getForwardLimit(),
+                motor.getReverseLimit()
+        );
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java:38]: move to a function

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java [hunk 2]
```diff
@@ -46,42 +67,24 @@ public void resetAngle(Rotation2d position) {
...
-                        targetAngle.getRotations(),
-                        0.0,
-                        true,
-                        PIVOT_FEED_FORWARD.calculate(targetAngle.getRadians(), 0),
-                        STAND_IN_PLACE_PID_SLOT,
-                        true,
-                        true,
-                        true
-                )
+                new PositionVoltage(targetAngle.getRotations())
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java:70]: create a position voltage at the top of the class like the MK4iSwerveModule class

File: src/main/java/edu/greenblitz/robotName/RobotConstants.java [hunk 3]
```diff
@@ -23,6 +23,8 @@ public static class Motors {
 			public final static double NEO_TICKS_PER_ROTATION = 42;
 			public static final double NEO_PHYSICAL_TICKS_TO_RADIANS = SPARKMAX_TICKS_PER_RADIAN / NEO_TICKS_PER_ROTATION; //do not use unless you understand the meaning
 			public final static double FALCON_REVOLUTIONS_PER_RADIAN = 2 * Math.PI;
+			public final static double SIGNAL_FREQUENCY_HERTZ = 100;
+			public final static double REACHED_SWITCH_LIMIT = 0;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/RobotConstants.java:27]: rename to isSwitchClosed

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivotConstants.java [hunk 4]
```diff
@@ -1,70 +1,76 @@
...
 
     public static final int MOTION_MAGIC_PID_SLOT = 1;
-    public static final int STAND_IN_PLACE_PID_SLOT = 1;
 
-    public static final MotionMagicConfigs MOTION_MAGIC_CONFIGS = new MotionMagicConfigs();
-    static {
-        MOTION_MAGIC_CONFIGS.MotionMagicAcceleration = 1;
-        MOTION_MAGIC_CONFIGS.MotionMagicCruiseVelocity = 2;
-        MOTION_MAGIC_CONFIGS.MotionMagicJerk = 4;
+    public static final int PID_MOTION_SLOT = 0;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivotConstants.java:28]: why MOTION?

**Review by bossyshrimp68 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivotConstants.java [hunk 5]
```diff
@@ -1,70 +1,76 @@
...
+        CURRENT_LIMITS_CONFIGS.SupplyCurrentLimit = 40;
         CURRENT_LIMITS_CONFIGS.SupplyCurrentLimitEnable = true;
-        CURRENT_LIMITS_CONFIGS.SupplyCurrentThreshold = 3;
-        CURRENT_LIMITS_CONFIGS.SupplyTimeThreshold = 4;
     }
 
     public static final SoftwareLimitSwitchConfigs SOFTWARE_LIMIT_SWITCH_CONFIGS = new SoftwareLimitSwitchConfigs();
     static {
-        SOFTWARE_LIMIT_SWITCH_CONFIGS.ForwardSoftLimitEnable = true;
+        SOFTWARE_LIMIT_SWITCH_CONFIGS.ForwardSoftLimitEnable = false;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivotConstants.java:58]: why false? the limit should be enabled

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java [hunk 6]
```diff
@@ -1,27 +1,53 @@
...
+        optimizeCanBus();
+
+        absoluteEncoder = new DutyCycleEncoder(ABSOLUTE_ENCODER_CHANNEL);
+
+        resetAngle(Rotation2d.fromRotations(absoluteEncoder.getAbsolutePosition()));
+
+        pidController = new PositionVoltage(motor.getPosition().getValue());
     }
 
+    public void optimizeCanBus(){
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java:37]:
```suggestion
    public void optimizeCANBusUtilization(){
```

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java [hunk 7]
```diff
@@ -1,27 +1,53 @@
...
+import static edu.greenblitz.robotName.RobotConstants.General.Motors.IS_SWITCH_CLOSED;
 import static edu.greenblitz.robotName.subsystems.shooter.Pivot.FalconPivot.FalconPivotConstants.*;
 
 public class FalconPivot implements IPivot {
 
     private TalonFX motor;
+    
+    private DutyCycleEncoder absoluteEncoder;
 
+    private PositionVoltage pidController;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivot.java:22]: rename to positionVoltage
```suggestion
    private PositionVoltage positionVoltage;
```

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivotConstants.java [hunk 8]
```diff
@@ -1,55 +1,54 @@
...
 
-    public static final ClosedLoopRampsConfigs CLOSED_LOOP_RAMPS_CONFIGS = new ClosedLoopRampsConfigs();
+    public static final MotionMagicConfigs MOTION_MAGIC_CONFIGS = new MotionMagicConfigs();
     static {
-        CLOSED_LOOP_RAMPS_CONFIGS.DutyCycleClosedLoopRampPeriod = 1;
-        CLOSED_LOOP_RAMPS_CONFIGS.TorqueClosedLoopRampPeriod = 2;
-        CLOSED_LOOP_RAMPS_CONFIGS.VoltageClosedLoopRampPeriod = 3;
+        MOTION_MAGIC_CONFIGS.MotionMagicAcceleration = kA;
+        MOTION_MAGIC_CONFIGS.MotionMagicCruiseVelocity = kV;
     }
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/FalconPivot/FalconPivotConstants.java:44]: remove

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

**Review by NoamRosenberg08 — APPROVED** [review]

**Review by yonnbro — APPROVED** [review]

**Pull request merged**